api/parts/jobs/handle.js

```diff
@@ -112,6 +113,38 @@ class Handle {
         return ipc;
     }
 
+
+    /**
+     * Suspend a job from UI
+     * @param {object} params - params object with job id
+     * @returns {object} - result of the operation
+     **/
+    async suspendJob(params) {
+        try {
+            let jobStatus = params.qstring.suspend ? job.STATUS.SUSPENDED : job.STATUS.SCHEDULED;
+            const result = await common.db.collection('jobs').findOne({_id: common.db.ObjectID(params.qstring.id), status: {$ne: jobStatus}});
+
+            if (result) {
+                const currentStatus = result.status;
+
+                if (currentStatus !== jobStatus) {
+                    await common.db.collection('jobs').updateOne({_id: result._id}, {$set: {status: jobStatus}});
```

Comment:
findOneAndUpdate was correct here, by separating it into 2 queries you didn't fix it @pnrgenc